### ISSUE TITLE
Design decision: Single level data where variable lengths are not equal

### ISSUE THREAD

**Issue by aaronrudkin** [DeclareDesign/fabricatr#16]
Considering just single level data, suppose a user makes the call: `fabricate(N = 4, test1=runif(6))`. Then they get an error correctly noting that N and the data variable imply differing data lengths.

But if they enter an N that is an even divisor of length(variable), the ID numbers get recycled: `fabricate(N = 3, test1=runif(6))` -- the resulting ID values are 1, 2, 3, 1, 2, 3.

Likewise, if the actual data are of even divisible length, the data get recycled:
`fabricate(N = 6, test1=runif(6), test2=runif(3))` -- the resulting test2 variables get recycled.

This suggests two questions we should answer:

1. Should it be possible to provide multiple columns of data with differing lengths?

2a. If so, should it be possible to provide an N different than the maximum length of the generated variables?
2b. If not, it surely sounds as though we should error if the expected N is not the length of the data rows.

We might imagine that users are typically going to use `n=N` in an argument, but there are perhaps some cases where they might not, and so we should have either a clear error message or explicitly document that this behaviour is permitted and expected.

**Comment by nfultz** [comment]
If a user wanted to recycle values, they could use `rep()` explicitly - I vote for making this an error, or at the minimum a warning with ominous wording.

**Comment by graemeblair** [comment]
Error sounds good to me, too

**Comment by graemeblair** [comment]
@aaronrudkin when I run the two code blocks above, I don't get an error -- should this issue be closed?

i.e.

```
fabricate(N = 6, test1=runif(6), test2=runif(3))
```

**Comment by aaronrudkin** [comment]
Hi guys. I thought I had implemented an error on this. It possibly got lost when Neal refactored the main fabricate and I pulled his changes into my feature branch, but I'm not sure. Oops! Regardless, the rewrite of fabricate will moot it as it will enforce that N is equal across all data within a level, and across connections between levels where it makes sense. I'll leave this one open for now as a reference for all the data sanity issues.
